tls/s2n_security_policies.c

```diff
@@ -891,6 +900,30 @@ int s2n_config_set_cipher_preferences(struct s2n_config *config, const char *ver
     return 0;
 }
 
+static bool s2n_cipher_suite_uses_chacha20_alg(struct s2n_cipher_suite *cipher_suite) {
+    if (!cipher_suite) {
+        return false;
+    }
+
+    /* If chacha20-poly1305 is available, then we can directly check the record_alg */
+    if (s2n_chacha20_poly1305.is_available()) {
+        return cipher_suite->record_alg && cipher_suite->record_alg->cipher == &s2n_chacha20_poly1305;
+    } 
+
+    /* Else chacha20-poly1305 is not available and so the cipher_suite's record_alg is never set. In order to determine
```

Comment:
```suggestion
    /* If s2n_chacha20_poly1305 is not available the record_alg->cipher will be null, so that field can't be used to 
     * check if this ciphersuite is ChaCha20. However we can iterate over the ciphersuite's list of record algorithms,
     * as a ChaCha20 ciphersuite will always have a ChaCha20 record algorithm in that list. */
```
Bleh. This is difficult to explain. Wouldn't it be easier to instead check if the ciphersuite.name contains "CHACHA20-POLY1305"? Did we consider that?

Reply:
Yeah I thought about doing that in the previous PR as well. String manipulation seems more risky than just directly comparing against the record_alg pointers. The comment change makes sense though, I'll update that.